src/plugins/data/public/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -30,6 +30,7 @@ import {
 
 import { ES_FIELD_TYPES, KBN_FIELD_TYPES, IIndexPattern, IFieldType } from '../../../common';
 
+import { META_FIELDS_SETTING } from '../../../../discover/common';
```

Comment:
It hasn't been mandatory so far. But it will be soon https://github.com/elastic/kibana/issues/64843

Reply:
@pgayvallet @restrry Mind if we remove this cross imports for now, collect them in a separate issue to migrate them to the owner plugin contract later on?

Reply:
FYI: I've moved META_FIELDS_SETTING + DOC_HIGHLIGHT_SETTING to the data plugin, since it's only used there

Reply:
@pgayvallet @restrry I took a look at `src/plugins/saved_objects`, that contains 

```TypeScript
export const PER_PAGE_SETTING = 'savedObjects:perPage';
export const LISTING_LIMIT_SETTING = 'savedObjects:listingLimit';
```

Plugins like dashboard, discover, visualize, gis just use helper functions of this plugins like this:
```TypeScript
import {
  createSavedObjectClass,
  SavedObject,
  SavedObjectKibanaServices,
} from '../../../saved_objects/public';
```

The current use of the constants is
```
import { PER_PAGE_SETTING } from '../../../saved_objects/common';
```

If I understand #64843 this should be fine, if `saved_objects` is declared as dependency?
Because refactoring to use an a setting accessor would mean to change the way those plugins use the `saved_objects ` plugin.

Reply:
> Because refactoring to use an a setting accessor would mean to change the way those plugins use the saved_objects plugin.

The main point here is to change the way **how** plugins access the SO value.
> But accessing a setting owned by another plugin would be something like
>const externalSetting = startDeps.dependencyPlugin.settings.getMaxBuckets()

Reply:
This for your input, I've followed @pgayvallet advise 👍